projects/pt1/python/torch_mlir_e2e_test/registry.py

```diff
@@ -15,7 +15,9 @@
 _SEEN_UNIQUE_NAMES = set()
 
 
-def register_test_case(module_factory: Callable[[], torch.nn.Module]):
+def register_test_case(
+    module_factory: Callable[[], torch.nn.Module], timeout: int = 600
+):
```

Comment:
> A default 600s timeout for all tests

10 minutes for an individual unit test seems like a _very_ long timeout. How about just 1 minute? or 3 minutes? What are the longest test times today (on typical hardware)?

(see also my other comment about units - without explicit units one might think this was measured in milliseconds, nanoseconds, or minutes)

Reply:
fixed the naming too. keeping 600s but will try halving it until cases start failing.

Reply:
Most test runners have a way to print the duration of each test case, sorted from longest to shortest. That's what I'd do if it were available here, instead of that sort of bisection search.